src/test/regress/expected/coordinator_tables.out

```diff
@@ -0,0 +1,230 @@
+NOTICE:  executing the command locally: SELECT worker_apply_shard_ddl_command (1504024, 'coordinator_tables_test_schema', 'CREATE TABLE coordinator_tables_test_schema.coordinator_table_2 (a integer NOT NULL)');SELECT worker_apply_shard_ddl_command (1504024, 'coordinator_tables_test_schema', 'ALTER TABLE coordinator_tables_test_schema.coordinator_table_2 OWNER TO postgres');SELECT worker_apply_shard_ddl_command (1504024, 'coordinator_tables_test_schema', 'ALTER TABLE coordinator_tables_test_schema.coordinator_table_2 ADD CONSTRAINT coordinator_table_2_pkey PRIMARY KEY (a)')
+ create_coordinator_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE TABLE local_table (a int primary key);
+CREATE TABLE distributed_table (a int primary key);
+SELECT create_distributed_table('distributed_table', 'a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE TABLE reference_table (a int primary key);
+SELECT create_reference_table('reference_table');
+ create_reference_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- any foreign key between coordinator tables and other tables cannot be set for now
+-- each should error out (for now meaningless error messages)
+-- between coordinator tables
+ALTER TABLE coordinator_table_1 ADD CONSTRAINT fkey_c_to_c FOREIGN KEY(a) references coordinator_table_2(a);
+ERROR:  cannot create foreign key constraint since relations are not colocated or not referencing a reference table
+-- between coordinator tables and reference tables
+ALTER TABLE coordinator_table_1 ADD CONSTRAINT fkey_c_to_ref FOREIGN KEY(a) references reference_table(a);
+ERROR:  cannot create foreign key constraint since relations are not colocated or not referencing a reference table
+ALTER TABLE reference_table ADD CONSTRAINT fkey_ref_to_c FOREIGN KEY(a) references coordinator_table_1(a);
+ERROR:  cannot create foreign key constraint since foreign keys from reference tables to distributed tables are not supported
+-- between coordinator tables and distributed tables
+ALTER TABLE coordinator_table_1 ADD CONSTRAINT fkey_c_to_dist FOREIGN KEY(a) references distributed_table(a);
+ERROR:  cannot create foreign key constraint since relations are not colocated or not referencing a reference table
+ALTER TABLE distributed_table ADD CONSTRAINT fkey_dist_to_c FOREIGN KEY(a) references coordinator_table_1(a);
+ERROR:  cannot create foreign key constraint since relations are not colocated or not referencing a reference table
+-- between coordinator tables and local tables
+ALTER TABLE coordinator_table_1 ADD CONSTRAINT fkey_c_to_local FOREIGN KEY(a) references local_table(a);
+ERROR:  cannot create foreign key constraint
+ALTER TABLE local_table ADD CONSTRAINT fkey_local_to_c FOREIGN KEY(a) references coordinator_table_1(a);
+ERROR:  relation local_table is not distributed
```

Comment:
We should improve this error message (not now, when we implement foreign key support)

Reply:
exactly